src/System.Net.Http/tests/FunctionalTests/HttpClientHandlerTest.Finalization.cs

```diff
@@ -57,5 +57,54 @@ public async Task IncompleteResponseStream_ResponseDropped_CancelsRequestToServe
                 }));
             }
         }
+
+        [Fact]
+        public async Task ClientNotDisposed_ServerReadsFromStream_ServerHangs()
+        {
+            Diagnostics.Debugger.Launch();
+            using (HttpClient client = new HttpClient())
+            {
+                await LoopbackServerFactory.CreateClientAndServerAsync(async url =>
+                {
+                    var stream = await client.GetStreamAsync(url);
+                    stream.Close();
+                    //client.Dispose(); Uncomment this to close the request stream and make the test succeed
+                },
+                server => server.AcceptConnectionAsync(async con =>
+                {
+                    await con.SendResponseAsync(content: "Response");
+
+                    byte[] buf = new byte[1024];
+                    int count = 0;
+                    do
+                    {
+                        count = con.Stream.Read(buf, 0, buf.Length);
+                    } while (count > 0);
+                }));
+            }
+        }
+
+        [Fact]
+        public async Task ClientNotDisposed_ClientSendsSecondRequest_ClientHangs()
+        {
+            using (HttpClient client = new HttpClient())
+            {
+                var requestRead = new AutoResetEvent(false);
+                await LoopbackServerFactory.CreateClientAndServerAsync(async url =>
+                {
+                    var stream = await client.GetStreamAsync(url);
+                    stream.Close();
+                    requestRead.Set();
+                    stream = await client.GetStreamAsync(url);
```

Comment:
now there is no server handler at this point. So this will timeout as nobody is answering.